src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -427,12 +422,14 @@ public function checkCase(\ReflectionClass $refl, $file, $class)
         ) {
             return [substr($tail, -$tailLen + 1), substr($real, -$tailLen + 1), substr($real, 0, -$tailLen + 1)];
         }
+
+        return null;
     }
 
     /**
      * `realpath` on MacOSX doesn't normalize the case of characters.
      */
-    private function darwinRealpath($real)
+    private function darwinRealpath($real): string
```

Comment:
`string $real`?

Reply:
this method is called only once, a few lines above via `$real = $this->darwinRealpath($real);`. `$real` in this case comes from `$real = $refl->getFileName();`. the `getFileName`-method of a `ReflectionClass` on the other hand can return either a string or false, hence the parameter for the `darwinRealpath` can not be any more explicit.

Reply:
`false` is feasible for `string $real` signature, it's casted to empty string `''` automatically, so it's fine type-hinting it with `string` here.

Reply:
you're right, my bad. fixed it, thanks for the hint!